File: bench/iops/loop.py
#!/usr/bin/env python3
import os
import time

n = int(os.environ.get("IOPS_N", "1000000"))
a = 0
t0 = time.perf_counter_ns()
while a < n:
    a = a + 1
t1 = time.perf_counter_ns()
if a != n:
    raise SystemExit(f"python: a={a} want {n}")
ns = t1 - t0
print(f"python n={n} ns={ns} ns/iter={ns / n:.3f} a={a}")

File: bench/iops/loop.rs
// #204 floor: rustc -O while black_box(a) < black_box(n) { a = black_box(a)+1 }
use std::hint::black_box;
use std::time::Instant;

fn main() {
    let n: i64 = std::env::var("IOPS_N")
        .ok()
        .and_then(|s| s.parse().ok())
        .unwrap_or(1_000_000);
    let mut a: i64 = 0;
    let t0 = Instant::now();
    while black_box(a) < black_box(n) {
        a = black_box(a) + 1;
    }
    let ns = t0.elapsed().as_nanos();
    if a != n {
        eprintln!("rust: a={a} want {n}");
        std::process::exit(1);
    }
    println!(
        "rust n={n} ns={ns} ns/iter={:.3} a={a}",
        ns as f64 / n as f64
    );
}
